src/value/generator.rs

```diff
@@ -197,6 +135,170 @@ pub trait BaseGenerator {
         self.write_char(b'"')
     }
 
+    #[cfg(target_feature = "neon")]
```

Comment:
this should live somewhere in the `neon/` directory. Or more generally since there are now 3 different implementation of write_string they all should be in their respective directories.